springfox-spring-web/src/main/java/springfox/documentation/spring/web/plugins/JavaDocPropertiesReader.java

```diff
@@ -0,0 +1,77 @@
+/*
+ *
+ *  Copyright 2015 the original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ */
+package springfox.documentation.spring.web.plugins;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.Properties;
+
+import javax.annotation.PostConstruct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/**
+ *
+ */
+@Component
+public class JavaDocPropertiesReader {
```

Comment:
We don't need to manually read property files here. Spring has a much easier way to read properties loaded into the environment. So for e.g. In your configuration of this module you might do something like this

```
@Configuration
@PropertySource("classpath:springfox.javadoc.properties")
public class JavadocModuleConfiguration {
}
```

and you could easily access these properties using the spring `Environment`. Infact we already do something very similar in [`ApiModelPropertiesBuilder`](https://github.com/springfox/springfox/blob/master/springfox-swagger-common/src/main/java/springfox/documentation/swagger/schema/ApiModelPropertyPropertyBuilder.java#L63) using [`DescriptionResolver`](https://github.com/springfox/springfox/blob/master/springfox-spring-web/src/main/java/springfox/documentation/spring/web/DescriptionResolver.java). See here for [an example](https://github.com/springfox/springfox/blob/master/springfox-swagger-common/src/main/java/springfox/documentation/swagger/schema/ApiModelProperties.java#L103-L118).

Secondly the property file need not be in the META-INF directory. It can be in the resources directory as well, which makes it easy for spring to find